tests/integration/mempool/test_helper.go

```diff
@@ -0,0 +1,153 @@
+package mempool
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/crypto/tmhash"
+
+	"github.com/cosmos/evm/testutil/integration/base/factory"
+	"github.com/cosmos/evm/testutil/keyring"
+	evmtypes "github.com/cosmos/evm/x/vm/types"
+
+	sdkmath "cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// Constants
+const (
+	TxGas = 100_000
+)
+
+// createCosmosSendTransactionWithKey creates a simple bank send transaction with the specified key
+func (s *IntegrationTestSuite) createCosmosSendTx(key keyring.Key, gasPrice *big.Int) sdk.Tx {
+	feeDenom := "aatom"
+
+	fromAddr := key.AccAddr
+	toAddr := s.keyring.GetKey(1).AccAddr
+	amount := sdk.NewCoins(sdk.NewInt64Coin(feeDenom, 1000))
+
+	bankMsg := banktypes.NewMsgSend(fromAddr, toAddr, amount)
+
+	gasPriceConverted := sdkmath.NewIntFromBigInt(gasPrice)
+
+	txArgs := factory.CosmosTxArgs{
+		Msgs:     []sdk.Msg{bankMsg},
+		GasPrice: &gasPriceConverted,
+	}
+	tx, err := s.factory.BuildCosmosTx(key.Priv, txArgs)
+	s.Require().NoError(err)
+
+	fmt.Printf("DEBUG: Created cosmos transaction successfully\n")
+	return tx
+}
+
+// createEVMTransaction creates an EVM transaction using the provided key
+func (s *IntegrationTestSuite) createEVMValueTransferTx(key keyring.Key, nonce int, gasPrice *big.Int) sdk.Tx {
+	to := s.keyring.GetKey(1).Addr
+
+	if nonce < 0 {
+		s.Require().NoError(fmt.Errorf("nonce must be non-negative"))
+	}
+
+	ethTxArgs := evmtypes.EvmTxArgs{
+		Nonce:    uint64(nonce),
+		To:       &to,
+		Amount:   big.NewInt(1000),
+		GasLimit: TxGas,
+		GasPrice: gasPrice,
+		Input:    nil,
+	}
+	tx, err := s.factory.GenerateSignedEthTx(key.Priv, ethTxArgs)
+	s.Require().NoError(err)
+
+	return tx
+}
+
+// createEVMContractDeployTx creates an EVM transaction for contract deployment
+func (s *IntegrationTestSuite) createEVMContractDeployTx(key keyring.Key, gasPrice *big.Int, data []byte) sdk.Tx {
+	ethTxArgs := evmtypes.EvmTxArgs{
+		Nonce:    0,
+		To:       nil,
+		Amount:   nil,
+		GasLimit: TxGas,
+		GasPrice: gasPrice,
+		Input:    data,
+	}
+	tx, err := s.factory.GenerateSignedEthTx(key.Priv, ethTxArgs)
+	s.Require().NoError(err)
+
+	return tx
+}
+
+// checkTxs call abci CheckTx for multipile transactions
+func (s *IntegrationTestSuite) checkTxs(txs []sdk.Tx) error {
+	for _, tx := range txs {
+		if err := s.checkTx(tx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// checkTxs call abci CheckTx for a transaction
+func (s *IntegrationTestSuite) checkTx(tx sdk.Tx) error {
+	txBytes, err := s.network.App.GetTxConfig().TxEncoder()(tx)
+	if err != nil {
+		return fmt.Errorf("failed to encode cosmos tx: %w", err)
+	}
+
+	_, err = s.network.App.CheckTx(&abci.RequestCheckTx{
```

Comment:
Let's make sure that no changes are actually persisted to the context (i.e. sequence incrementing) after a simple CheckTx is done. I a normal environment, changes would actually save until the finalization anyways.